### DISCUSSION TITLE
New release candidate: v1.5.5-rc.1

### DISCUSSION THREAD

**Discussion by fmartingr** [go-shiori/shiori#637] [Announcements]
Hello everyone!

I have cut a new release candidate with a few changes that had been on master for some time: [v1.5.5-rc.1](https://github.com/go-shiori/shiori/releases/tag/v1.5.5-rc.1).

Feel free to download and try it out for a while, and if you experience any issues report them so we can fix those before the final release.

Huge thanks to all contributors!

### Features
* feat: add SHIORI_PG_SSLMODE by @refi64 in https://github.com/go-shiori/shiori/pull/536
* feat: Initial Ebook support by @Monirzadeh in https://github.com/go-shiori/shiori/pull/623
* feat: improve bookmark image detection by @Katarn in https://github.com/go-shiori/shiori/pull/554
* feat: Add log args for root command by @chain710 in https://github.com/go-shiori/shiori/pull/565

### Fixes
* fix: Use UTC for cookie expiry date in backend by @MoX46 in https://github.com/go-shiori/shiori/pull/566
* fix: browser ext cannot archive and update bookmark. by @IvanLi-CN in https://github.com/go-shiori/shiori/pull/560
* fix(frontend): prevent archive header covering up the archive content by @sdvcrx in https://github.com/go-shiori/shiori/pull/562
* fix direction in RTL language by @Monirzadeh in https://github.com/go-shiori/shiori/pull/615
* fix(log): record user real ip from headers by @Tomilla in https://github.com/go-shiori/shiori/pull/603
* fix: thumb image will be not accesable anymore if it is private #578 by @Monirzadeh in https://github.com/go-shiori/shiori/pull/627
* fix: prevent bookmark overwrite on archive download by @fmartingr in https://github.com/go-shiori/shiori/pull/636

### Dependencies & CI
* Update dependencies by @fmartingr 
  * https://github.com/go-shiori/shiori/pull/616
  * https://github.com/go-shiori/shiori/pull/567
* deps: update to go 1.20 + dependencies. Fixes CI by @fmartingr in https://github.com/go-shiori/shiori/pull/628

**Comment by omnisci3nce** [comment]
Thanks for keeping this project going. Awesome work.

**Comment by blogbourri** [comment]
When I updated my docker install, the update broke it so I couldn't login, giving an "sql: no rows in result set (500)" error.

**Reply by fmartingr** [reply]
From which version did you upgrade? Where do you have your shiori data folder?

I updated my production shiori from 1.5.4 without issues.

**Reply by blogbourri** [reply]
from 1.5.4, i was just using the sqlite database with i think default settings. i got impatient and wiped the install after (was meaning to switch over to mysql anyways), so unfortunately i can't go back and figure out what caused it. if it doesn't happen for anyone else might have been a configuration error on my end (pretty new to self-hosting). i did run into a single bug for the ebook export, i'll do a proper bug report for that one in a bit. very excited to have a working ebook export, and it worked great for all but one archive.

**Reply by fmartingr** [reply]
How do you create the container? Did you delete it before the upgrade you say? 

I cant guarantee that no bugs are present, but upgrading should be non-destructive, since any database error stops the application from running before anything gets committed to disk. 

My upgrade path was the same as yours (same database as well). I plan to have proper install/upgrade instructions, along with a backup reminder as well. I hope you didnt lose many data. The other person I recall had a similar issue, his database was stored on a different path. #291

**Reply by blogbourri** [reply]
The upgrade when I had the error was handled by watchtower, which should exactly recreate all the same settings as the prior install. I did lose all my bookmarks, slight bummer. I probably could have recovered them if I'd been patient enough to try and figure out what happened, so that's partially on me.

Thinking about it, it could have just been because the database got wiped when the upgrade happened, if it was stored in the same directory, so that would probably be a configuration error on my end? I never figured out where the database was stored before.

**Reply by fmartingr** [reply]
Im not familiar with watchtower, but the only thing that comes to mind is that it didn't create a volume for the data shiori stores. Using an ephemeral volume would mean that any image upgrade (or container restart) would wipe the database.

**Comment by eitau** [comment]
Heads up: title is never retrieved
https://github.com/go-shiori/shiori/blob/v1.5.5-rc.1/internal/webserver/handler-api.go#L320-L322
```go
// Make sure bookmark's title not empty
if book.Title == "" {
    book.Title = book.URL
}
// ^ because if this
// …
        bookmark, err := downloadBookmarkContent(book, h.DataDir, r, book.Title != "", book.Excerpt != "")
//                                                                   ^ it's always non-empty
// …
    book, err = downloadBookmarkContent(book, h.DataDir, r, book.Title != "", book.Excerpt != "")
//                                                          ^ it's always non-empty
```
